### PR TITLE
bpo-41490: ``path`` and ``contents`` to aggressively close handles

### PR DESCRIPTION
**Pull request by jaraco**
As found in importlib_resources 3.1.1.

https://bugs.python.org/issue41490

Automerge-Triggered-By: GH:jaraco

### CODE REVIEW

**Comment by jaraco** [comment]
I need to investigate why the zip test data changed.

**Comment by jaraco** [comment]
I see I refreshed those zips in python/importlib_resources@93428774ba6ede77709babe2af5c0199a74a3872, and this is the first sync since.

**Comment by jaraco** [comment]
Oh, that's interesting. `test_contents_does_not_keep_open` still fails here where it doesn't on `importlib_resources`.

**Comment by jaraco** [comment]
I don't want to apply the fix at `ZipReader` because I want `ZipReader.contents` to evaluate lazily. The compatibility only needs to be applied to `contents()`.

File: Lib/test/test_importlib/test_resource.py [hunk 1]
```diff
@@ -162,5 +166,80 @@ def test_namespaces_cannot_have_resources(self):
...
+        c = resources.is_resource('ziptestdata', 'binary.file')
+        self.zip_path.unlink()
+        del c
+
+    def test_is_resource_failure_does_not_keep_open(self):
+        c = resources.is_resource('ziptestdata', 'not-present')
+        self.zip_path.unlink()
+        del c
+
+    @unittest.skip("Desired but not supported.")
```
Comment (jaraco) [inline Lib/test/test_importlib/test_resource.py:220]: I should call out that this test is being skipped because it proved too difficult to support.

Comment (zooba) [reply Lib/test/test_importlib/test_resource.py:220]: This one is but the following one is not?

Comment (jaraco) [reply Lib/test/test_importlib/test_resource.py:220]: Right. I found a way to support this use-case, but only by not supporting the latter... and the latter was more important to certifi.

**Comment by jaraco** [comment]
I'd like to proceed with merging this change. We can iterate or revert later as needed.

**Pull request merged**